primaryHandler.go

```diff
@@ -267,68 +267,74 @@ func fixV2Duration(getLogger ancla.GetLoggerFunc, config ancla.TTLVConfig) (alic
-						v2ErrEncode(w, logger,
-							fmt.Errorf("%w: %v must be of type %v", errFailedWebhookUnmarshal, e.Field, e.Type),
-							http.StatusBadRequest)
-						return
-					}
-					v2ErrEncode(w, logger, fmt.Errorf("%w: %v", errFailedWebhookUnmarshal, err),
+			var wr ancla.WebhookRegistration
+			err = json.Unmarshal(requestPayload, &wr)
+			if err != nil {
+				var e *json.UnmarshalTypeError
+				if errors.As(err, &e) {
+					v2ErrEncode(w, logger,
+						fmt.Errorf("%w: %v must be of type %v", errFailedWebhookUnmarshal, e.Field, e.Type),
 						http.StatusBadRequest)
 					return
 				}
+				v2ErrEncode(w, logger, fmt.Errorf("%w: %v", errFailedWebhookUnmarshal, err),
+					http.StatusBadRequest)
+				return
+			}
 
-				// check to see if the Webhook has a valid until/duration.
-				// If not, set the WebhookRegistration  duration to 5m.
-				webhook := wr.ToWebhook()
-				if webhook.Until.IsZero() {
-					if webhook.Duration == 0 {
-						wr.Duration = ancla.CustomDuration(config.Max)
-						w.Header().Add(v2WarningHeader,
-							fmt.Sprintf("Unset duration and until fields will not be accepted in v3, webhook duration defaulted to %v", config.Max))
-					} else {
-						durationErr := durationCheck(webhook)
-						if durationErr != nil {
-							wr.Duration = ancla.CustomDuration(config.Max)
-							w.Header().Add(v2WarningHeader,
-								fmt.Sprintf("Invalid duration will not be accepted in v3: %v, webhook duration defaulted to %v", durationErr, config.Max))
-						}
-					}
+			// check to see if the Webhook has a valid until/duration.
+			// If not, set the WebhookRegistration  duration to 5m.
+			webhook := wr.ToWebhook()
+			if webhook.Until.IsZero() {
+				if webhook.Duration == 0 {
+					wr.Duration = ancla.CustomDuration(config.Max)
+					w.Header().Add(v2WarningHeader,
+						fmt.Sprintf("Unset duration and until fields will not be accepted in v3, webhook duration defaulted to %v", config.Max))
 				} else {
-					untilErr := untilCheck(webhook)
-					if untilErr != nil {
-						wr.Until = time.Time{}
+					durationErr := durationCheck(webhook)
+					if durationErr != nil {
 						wr.Duration = ancla.CustomDuration(config.Max)
 						w.Header().Add(v2WarningHeader,
-							fmt.Sprintf("Invalid until value will not be accepted in v3: %v, webhook duration defaulted to 5m", untilErr))
+							fmt.Sprintf("Invalid duration will not be accepted in v3: %v, webhook duration defaulted to %v", durationErr, config.Max))
 					}
 				}
-
-				// put the body back in the request
-				body, err := json.Marshal(wr)
-				if err != nil {
-					v2ErrEncode(w, logger, fmt.Errorf("failed to recreate request body: %v", err), 0)
+			} else {
+				untilErr := untilCheck(webhook)
+				if untilErr != nil {
+					wr.Until = time.Time{}
+					wr.Duration = ancla.CustomDuration(config.Max)
+					w.Header().Add(v2WarningHeader,
+						fmt.Sprintf("Invalid until value will not be accepted in v3: %v, webhook duration defaulted to 5m", untilErr))
```

Comment:
should we have the same error code here as we had in line 317?

Reply:
No, because this time the Until value is wrong, not the duration value.  Since Until takes precedence over Duration, we set it to be empty/zero so that we can use the default duration without needing to calculate the exact Until value quite yet.